tests/test_provider_router.py

```
@@ -102,6 +102,50 @@ async def fake_post_chat(self, provider, payload, timeout_sec):
+            headers={"content-type": "application/json"},
+        )
+
+    monkeypatch.setattr(ProviderRouter, "_post_chat", fake_post_chat)
+    router = ProviderRouter(
+        [
+            ProviderConfig("fast-but-limited", "openai-compatible", "https://a/v1", api_key="k", default_model="a", priority=0),
+            ProviderConfig("backup", "openai-compatible", "https://b/v1", api_key="k", default_model="b", priority=10),
+        ]
+    )
+
+    # max_retries=2: the OLD behaviour would call fast-but-limited 3× before failover.
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)